schema.graphql

```diff
@@ -73,31 +79,63 @@ type VerseText {
   assessmentDetails: [AssessmentDetail]
 }
 
-enum AssessmentType {
+enum TextAnnotationType {
   SemanticSimilarity
   QAComprehensibility
   SubwordsPerSentence
+  VersePart
+  PartOfSpeech
+  NamedEntity
+}
+
+enum ReferenceAnnotationType {
+  SemanticDomain
+  KeyTerm
 }
 
 type Assessment {
   id: ID!
-  type: AssessmentType! @search(by: [exact])
+  type: TextAnnotationType! @search(by: [exact])
   bibleRevision: BibleRevision!
   reference: BibleRevision
   date: DateTime! @search
-  details: [AssessmentDetail!] @hasInverse(field: assessment)
+  details: [Annotation!] @hasInverse(field: assessment)
+}
+
+interface Annotation {
+  id: ID!
+  flag: Boolean @search
+  number: Float @search
+  tag: String @search
+  parent: [Annotation] @hasInverse(field: children)
+  children: [Annotation] @hasInverse(field: parent)
 }
 
-type AssessmentDetail {
+type TextAnnotation implements Annotation {
   id: ID!
-  assessment: Assessment!
-  assessmentMetrics: [AssessmentMetric!] @hasInverse(field: assessmentDetail)
-  verseTexts: [VerseText]
+  verseTexts: [VerseText!]
+  startPosition: Int
+  endPosition: Int
+  refVerseTexts: [VerseText]
+  refStartPosition: Int
+  refEndPosition: Int
+  flag: Boolean @search
+  number: Float @search
+  tag: String @search
+  parent: [Annotation] @hasInverse(field: children)
+  children: [Annotation] @hasInverse(field: parent)
+  assessment: [Assessment] @hasInverse(field: details)
+  type: TextAnnotationType!
 }
 
-type AssessmentMetric {
+type ReferenceAnnotation implements Annotation {
   id: ID!
-  assessmentDetail: AssessmentDetail!
+  reference: [Reference!]
   flag: Boolean @search
   number: Float @search
+  tag: String @search
+  parent: [Annotation] @hasInverse(field: children)
```

Comment:
Similar comment here. What is the use case for having annotations arranged hierarchically on the reference backbone?

Reply:
Similar to above. Let's leave this off for now, but document in the GH issue.